olive/model/__init__.py

```diff
@@ -637,29 +636,71 @@ def prepare_session(
     ):
         return self.load_model().eval()
 
+    def _resolve_io_config(self, io_config: Union[Dict[str, Any], IOConfig, str, Callable]) -> Dict[str, Any]:
+        """Resolve io_config to a dictionary.
+
+        If io_config is a string name or a callable, it will be called to get io_config.
+        """
+        if isinstance(io_config, dict):
+            # io_config is provided
+            return io_config
+
+        if isinstance(io_config, IOConfig):
+            # io_config is an IOConfig
+            return io_config.dict()
+
+        if isinstance(io_config, (str, Callable)):
+            # io_config is a string name or a callable
+            logger.debug(f"Calling {io_config} to get io_config")
+            user_module_loader = UserModuleLoader(self.model_script, self.script_dir)
+            io_config = user_module_loader.call_object(io_config, self)
+            return validate_config(io_config, IOConfig).dict()
+
+        return None
+
+    def get_io_config(self) -> Dict[str, Any]:
+        """Return io config of the model.
+
+        Priority: io_config > hf_config (using onnx_config)
+        """
+        io_config = None
+        if self.io_config:
+            # io_config is provided
+            io_config = self._resolve_io_config(self.io_config)
+        elif self.hf_config and self.hf_config.task and not self.hf_config.components:
+            # hf_config is provided
+            logger.debug("Using hf onnx_config to get io_config")
```

Comment:
Using hf *onnx*_config? It is pytorch model here.

Reply:
I found it in hf_utils. But somehow we may need to refine the name or logics here.
It looks weird to get `onnx config` under `get_io_config`.

Reply:
Like your comment below says, it's not related to this PR. It was already in use before this PR. The `onnx_config` comes from `transformers.onnx` https://github.com/huggingface/transformers/blob/main/src/transformers/onnx/config.py#L68. 

I have some plans to migrate from `transformers.onnx` to optimum since huggingface is not maintaining this module anymore. Maybe we can do the renaming then.